ui/app/adapters/watchable.js

```diff
@@ -125,17 +127,34 @@ export default class Watchable extends ApplicationAdapter {
 
       // Remove existing records that match this query. This way if server-side
       // deletes have occurred, the store won't have stale records.
-      store
+      const matchingStoreEntries = store
         .peekAll(type.modelName)
-        .filter((record) =>
-          queryParamsToAttrs.some(
-            (mapping) => get(record, mapping.attr) === query[mapping.queryParam]
-          )
-        )
-        .forEach((record) => {
-          removeRecord(store, record);
+        .filter((record) => {
+          return queryParamsToAttrs.some((mapping) => {
+            // Special consideration for * queries, like "All jobs with * namespace":
+            // entities will generally have "default" instead of "*" as their value for these sorts of properties.
+            const queryValue =
+              query[mapping.queryParam] === '*'
+                ? 'default'
+                : query[mapping.queryParam];
+            return (
+              get(record, mapping.attr) === queryValue ||
+              get(get(record, mapping.attr), 'id') === queryValue
+            );
```

Comment:
This is a great refactoring. I'd like to discuss this further and how it'll impact the work for [15008](https://github.com/hashicorp/nomad/issues/15008) to unblock [14787](https://github.com/hashicorp/nomad/issues/14787).